File: JavaExemplos/TiposJogosEnumJavaExemplo/src/com/exemplo/TipoJogo.java
package com.exemplo;

public enum TipoJogo {
	ACAO("Ação"), AVENTURA("Aventura"), RPG("Role-playing game");

	private String descricao;

	private TipoJogo(String descricao) {
		this.descricao = descricao;
	}

	public String getDescricao() {
		return descricao;
	}
}

File: CSExemplos/1TiposJogosEnumCsSwitchExemplo/TiposJogosComSwitch/TiposJogosComSwitch/TipoJogo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TiposJogosComSwitch
{
    public enum TipoJogo
    {
        ACAO, AVENTURA, RPG
    }

    public static class TipoJogoDescricao
    {
        public static string GetDescricao(this TipoJogo tipoJogo)
        {
            switch (tipoJogo)
            {
                case TipoJogo.ACAO:
                    return "Ação";
                case TipoJogo.AVENTURA:
                    return "Aventura";
                case TipoJogo.RPG:
                    return "Role-playing game";
            }

            throw new Exception("Descrição não encontrada");
        }
    }
}
